lib/rules/require-unicode-regexp.js

```diff
@@ -85,22 +139,48 @@ module.exports = {
                     const pattern = getStringIfConstant(patternNode, scope);
                     const flags = getStringIfConstant(flagsNode, scope);
 
-                    if (!flagsNode || (typeof flags === "string" && !flags.includes("u") && !flags.includes("v"))) {
+                    let missingFlag = !flagsNode;
+
+                    if (typeof flags === "string") {
+                        missingFlag = checkFlags(requireFlag, flags);
+                    }
+
+                    if (missingFlag) {
                         context.report({
-                            messageId: "requireUFlag",
+                            messageId: requireFlag === "v" ? "requireVFlag" : "requireUFlag",
                             node: refNode,
-                            suggest: typeof pattern === "string" && isValidWithUnicodeFlag(context.languageOptions.ecmaVersion, pattern)
+                            suggest: typeof pattern === "string" && isValidWithUnicodeFlag(context.languageOptions.ecmaVersion, pattern, requireFlag)
                                 ? [
                                     {
                                         fix(fixer) {
+                                            const replaceFlag = requireFlag ?? "u";
+
                                             if (flagsNode) {
                                                 if ((flagsNode.type === "Literal" && typeof flagsNode.value === "string") || flagsNode.type === "TemplateLiteral") {
+
+                                                    // Avoid replacing "u" in escapes like `\uXXXX`
+                                                    if (flagsNode.type === "Literal" && flagsNode.raw.includes("\\")) {
+                                                        return null;
+                                                    }
+
+                                                    // Avoid replacing "u" in expressions like "`${regularFlags}g`"
+                                                    if (flagsNode.type === "TemplateLiteral" && (
+                                                        flagsNode.expressions.length ||
+                                                        flagsNode.quasis.some(({ value: { raw } }) => raw.includes("\\"))
+                                                    )) {
+                                                        return null;
+                                                    }
```

Comment:
I've avoided the other checks when `flagsNodeText` doesn't contain `flag`, but the replacement of `flag` with `""` is I think harmless when `flagsNodeText` doesn't contain `flag` since it has no effect, and indeed the new flag does just get appended in such a case. I could add additional code without the `replace` but it seems to me a little verbose.

Reply:
Can we check if `flags` contains `flag`? In code like the following, it should be fine to add `v` to the end (we just shouldn't be replacing `u` character):

```js
/*eslint require-unicode-regexp: ["error", { "requireFlag": "v" }] */

new RegExp('foo', '\u0067'); // g flag


```

Reply:
I see, thanks. I've added a commit.